File: w02/d02/alexander_levine/pets/lib/pet.rb
class Pet

  attr_reader :name, :age, :owner

  def initialize(name, age, owner)
    @name = name
    @owner = owner
    @age = age
    @foods_eaten = []
  end

  def love_owner
    return "I love #{owner}!"
  end

  def eat(food)
    @foods_eaten.push(food)
  end

  def list_foods
    @foods_eaten.join(", ")
  end

end



# * Pet
#   * name - string
#   * owner - string, owner's name
#   * age - number
#   * foods_eaten - initialized to empty array
#   * love_owner() - return a string proclaiming love towards owner
#   * eat(food) - push 'food' into the foods_eaten array
#   * list_foods() - return a string containing all the eaten foods
# * Dog
#   * ear_type - A string (Eg. "Floopy", "Upright", "Stubby", "Shaggy")
#   * bark() - Returns a string with a bark in it (eg. "Woof woof")
# * Cat
#   * hours_slept - initialized to 0
#   * sleep() - increments hours_slept by 1
#   * judge_person(person) - Accepts a string 'person' and returns a judgmental string about that person

File: w10/d05/hoa_newton/inheritance/pet.js
function Pet(options) {
  this.name = options.name;
  this.owner = options.owner;
}
Pet.prototype.introduce = function () {
  return "Hi, My name is " + this.name + " and I belong to " + this.owener;
}

/////////////////////////////////////
function Cat ( name, owner, favFood ) {
  Pet.call(this, name, owner);
  this.favFood = favFood;
}
Cat.prototype = Object.create(Pet.prototype);
Cat.prototype.judgePerson = function ( name ) {
  return name + " is aweful! ";
}
Cat.prototype.introduce = function () {
  return "My name is " + this.name + " and this person "+this.owner+ " belongs to me.";
}

////////////////////////////////////
function Dog ( name, owner, earType ) {
  Pet.call(this, name, owner);
  this.earType = earType;
}
Dog.prototype = Object.create(Pet.prototype);
Dog.prototype.bark = function () {
  "Rrrrrrruffffffff Rrrrrrruffffffff";
}


